roles/deploy/tasks/share.yml

```diff
@@ -2,25 +2,41 @@
 - include: "{{ deploy_share_before | default('../hooks/example.yml') }}"
   tags: deploy-share-before
 
-- name: Ensure shared sources are present
+- name: Ensure shared sources are present -- directories
+  file:
+    path: "{{ deploy_helper.shared_path }}/{{ project_shared_file | ternary(item.src | dirname, item.src) }}"
+    state: "{{ project_shared_file | ternary('directory', item.type | default('directory')) | lower }}"
+    mode: "{{ project_shared_file | ternary('0755', item.mode | default('0755')) }}"
+  with_items: "{{ project_shared_children }}"
+
+- name: Ensure shared sources are present -- files
   file:
     path: "{{ deploy_helper.shared_path }}/{{ item.src }}"
-    state: "{{ item.type | default('directory') }}"
-    mode: "{{ item.mode | default('0755') }}"
+    state: touch
+    mode: "{{ item.mode | default('0644') }}"
+  with_items: "{{ project_shared_children }}"
+  when: project_shared_file
+
+- name: Ensure parent directories for shared paths are present
+  file:
+    path: "{{ deploy_helper.new_release_path }}/{{ item.path | default('') | dirname }}"
+    state: directory
   with_items: "{{ project_shared_children }}"
 
 - name: Ensure shared paths are absent
   file:
-    path: "{{ deploy_helper.new_release_path }}/{{ item.path }}"
+    path: "{{ deploy_helper.new_release_path }}/{{ item.path  | default('') }}"
     state: absent
   with_items: "{{ project_shared_children }}"
+  when: item.type | default('directory') | lower != 'absent'
 
 - name: Create shared symlinks
   file:
-    path: "{{ deploy_helper.new_release_path }}/{{ item.path }}"
+    path: "{{ deploy_helper.new_release_path }}/{{ item.path | default('') }}"
```

Comment:
The `absent` option allows an easy way for someone to remove something from `/shared` if it will no longer be used in the next deployment(s). Although it is not necessary to remove items, and although people could remove them manually, facilitating this `absent` option requires no extra tasks, so I thought it was convenient.

However, the `absent` option does require the the two tasks you've marked to include `when` conditions and `item.path | default('')`. Yes these two tasks will skip for items set to `absent`, but Ansible still attempts to template all the variables in the task. So, without a default for `item.path`, the task would fail when people omit `path` for items set to `absent`. 

If we keep this `absent` option, I would like people to be able to omit `path` because an item to be removed from `/shared` will have no `path` in the release directory, so the `path` parameter doesn't make sense to include.